pkg/gather/clusterconfig/clusterconfig.go

```diff
@@ -816,6 +822,161 @@ func GatherMachineSet(i *Gatherer) func() ([]record.Record, []error) {
 	}
 }
 
+// GatherInstallPlans collects Top x InstallPlans from all openshift namespaces.
+// Because InstallPlans have unique generated names, it groups them by namespace and the "template"
+// for name generation from field generateName.
+// It also collects Total number of all installplans and all non-unique installplans.
+//
+// The Operators-Framework api https://github.com/operator-framework/api/blob/master/pkg/operators/v1alpha1/installplan_types.go#L26
+//
+// Location in archive: config/installplans/
+func GatherInstallPlans(i *Gatherer) func() ([]record.Record, []error) {
+	return func() ([]record.Record, []error) {
+		var plansBatchLimit int64 = 500
+		cont := ""
+		recs := map[string]*collectedPlan{}
+		total := 0
+		opResource := schema.GroupVersionResource{Group: "operators.coreos.com", Version: "v1alpha1", Resource: "installplans"}
+
+		config, err := getAllNamespaces(i)
+		if errors.IsNotFound(err) {
+			return nil, nil
+		}
+		if err != nil {
+			return nil, []error{err}
+		}
+
+		// collect from all openshift* namespaces
+		for _, ns := range config.Items {
+			if !strings.HasPrefix(ns.Name, "openshift") {
```

Comment:
Do we want to filter installplans like that? I think user can install an OLM operator (which introduces new installplan AFAIK) to her/his own namespace.

Reply:
This is mainly to collect statistics of openshift installplans, to limit them, not all of them. It is a bit wider then original idea collecting installplans from one openshift namespace.

Reply:
OK. Thanks